### PR TITLE
#9853 Including IntegrationTestInitService.java

### PR DESCRIPTION
**Pull request by nollymar**
- Adding new singleton class to set up web environment for integration tests. Class name: IntegrationTestInitService.java
- Replacing call set up in ContentletBaseTest.java

### CODE REVIEW

File: test/com/dotmarketing/util/IntegrationTestInitService.java [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+import com.dotmarketing.servlets.test.ServletTestRunner;
+
+/**
+ * Sets up the web environment needed to execute integration tests without a server application
+ * Created by nollymar on 9/29/16.
+ */
+public class IntegrationTestInitService {
+    private static IntegrationTestInitService service = null;
+
+    private static boolean initCompleted;
```
Comment (jdotcms) [inline test/com/dotmarketing/util/IntegrationTestInitService.java]: Use AtomicBoolean to make it thread-safe

**Review by jdotcms — CHANGES_REQUESTED** [review]

File: test/com/dotmarketing/util/IntegrationTestInitService.java (hunk 1 above)
Comment (nollymar) [inline test/com/dotmarketing/util/IntegrationTestInitService.java]: Suggestion included in the PR

File: test/com/dotmarketing/util/IntegrationTestInitService.java [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+    private static AtomicBoolean initCompleted;
+
+    private IntegrationTestInitService() {
+        initCompleted = new AtomicBoolean(false);
+    }
+
+    public static IntegrationTestInitService getInstance() {
+        if (service == null) {
+            service = new IntegrationTestInitService();
+        }
```
Comment (freddyucv) [inline test/com/dotmarketing/util/IntegrationTestInitService.java]: Really it is not a thread safe way to implement a singleton.
here explain my point: 
http://crunchify.com/thread-safe-and-a-fast-singleton-implementation-in-java/

Comment (nollymar) [inline test/com/dotmarketing/util/IntegrationTestInitService.java]: I have applied changes to make the singleton class thread safe.
Thanks!

**Review by jdotcms — APPROVED** [review]

**Review by dsilvam — APPROVED** [review]

**Pull request merged**